src/js/launcher.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-env node */
+"use strict";
+var fluid  = require("infusion");
+var gpii   = fluid.registerNamespace("gpii");
+var kettle = fluid.registerNamespace("kettle");
+
+fluid.require("%kettle/lib/KettleConfigLoader.js");
+
+var yargs   = require("yargs");
+var path    = require("path");
+var process = require("process");
+
+fluid.registerNamespace("gpii.launcher");
+
+/**
+ *
+ * Merge options from (in order of precedence):
+ *
+ * 1. An options file, if supplied.
+ * 2. Environment variables.
+ * 3. Command line parameters.
+ *
+ * @param that - The `gpii.launcher` component itself.
+ *
+ */
+gpii.launcher.launchComponent = function (that) {
```

Comment:
After long discussion, we agreed to support both "pushing" information into components from the outside and "pulling" information from within options blocks.  The latest commits are an attempt at that.  Please review.